src/main/java/no/sikt/sws/PrefixStripper.java

```diff
@@ -62,39 +64,51 @@ public static String body(OpenSearchCommandKind commandKind,
             case ERROR:
-                return Builder.errorFromValues(workspacePrefix,responseBody);
+                return ErrorDto
+                    .fromResponse(responseBody)
+                    .strippedResponse(workspacePrefix);
             case CONTENT:
                 switch (commandKind) {
                     case ALIAS:
                     case MAPPING:
-                        return responseBody.replaceAll(regex, EMPTY_STRING);
+                        return IndexDto
+                            .fromResponse(responseBody)
+                            .strippedResponse(workspacePrefix);
                     case DOC:
-                        return Builder.docFromValues(workspacePrefix,responseBody);
+                        return DocDto
+                            .fromResponse(responseBody)
+                            .strippedResponse(workspacePrefix);
                     default:
-                        throw new IllegalStateException("Unexpected value: " + commandKind);
+                        throw new IllegalStateException(UNEXPECTED_VALUE + commandKind.name());
                 }
             case CONTENT_COLLECTION:
                 switch (commandKind) {
                     case BULK:
                     case MAPPING:
-                        return responseBody.replaceAll(regex, EMPTY_STRING);
+                        return responseBody.replaceAll(regex, EMPTY_STRING);    // stripping index names in body.
                     case INDEX:
-                        return Builder.indexFromValues(workspacePrefix,responseBody);
+                        return IndexesDto
+                            .fromResponse(responseBody)
+                            .strippedResponse(workspacePrefix);
                     case SEARCH:
-                        return Builder.searchFromValues(workspacePrefix,responseBody);
+                        return SearchDto
+                            .fromResponse(responseBody)
+                            .strippedResponse(workspacePrefix);
                     default:
-                        throw new IllegalStateException("Unexpected value: " + commandKind);
+                        throw new IllegalStateException(UNEXPECTED_VALUE + commandKind.name());
                 }
             default:
-                throw new IllegalStateException("Unexpected value: " + commandKind);
+                throw new IllegalStateException(UNEXPECTED_VALUE + commandKind.name());
         }
     }
 
-    private static void validateParameters(String workspacePrefix, String responseBody) {
-        if (responseBody == null || workspacePrefix == null) {
+    private static void validateParameters(String workspacePrefix, String responseBody,
+                                           OpenSearchCommandKind commandKind) {
+        if (responseBody == null || workspacePrefix == null || commandKind.compareTo(NOT_IMPLEMENTED) >= 0) {
```

Comment:
Can you instead implement a OpenSearchCommandKind.isValid() ? The order of the enums should not matter as it can be easily messed with by others later